csrc/kernel_cache.cpp

```diff
@@ -1379,7 +1380,9 @@ std::pair<LaunchParams, CompileParams> FusionKernelRuntime::getKernelConfig(
 
   // Check that the heuristics are matched, in the case of segmented fusion
   NVF_ERROR(!sg || scheduler_entry->heuristic() == sg->heuristic());
-  NVF_ERROR(executors_.at(group_id).isCompiled());
+  NVF_ERROR(
+      executors_.at(group_id).isCompiled() ||
+      executors_.at(group_id).isCompilationSkipped());
```

Comment:
Why does this check exist in the first place? The function itself doesn't seems to use `fusion_` or `kernel_`. It seems to belong to something like FusionExecutor::runFusion.

Reply:
I only see `scheduler_entry->params()->cparams.index_type` being updated in `compileFusion`.
Since this function is only called here before `runFusion`, we can move the checks inside `runFusion`.

Only if this function is used in isolation (which should probably not happen) it can have `index_type` vary based on compilation status